### PR TITLE
docs(support): add typescript 4.9 to support table 

### PR DESCRIPTION
**Pull request by rwaskiewicz**
this commit adds typescript 4.9 to the support table

~this PR has the commit for TS 4.8 support in it as well. I built this branch atop #934, to avoid merge conflicts in the future. this first commit will be gone on Monday after I merge the 4.8 PR.~ rebased following the v2.20 release

### CODE REVIEW

File: src/docs/reference/support-policy.md [hunk 1]
```diff
@@ -74,6 +74,8 @@ The table below describes recent versions of Stencil and the version of TypeScri
 
 | Stencil Version | TypeScript Version |
 |:---------------:|:------------------:|
+|     v2.21.0     |       v4.9.3       |
+|     v2.20.0     |       v4.8.4       |
```
Comment (tanner-reits) [inline src/docs/reference/support-policy.md:78]: Did we never update this for 4.8?

**Review by tanner-reits — APPROVED** [review]

**Comment by tanner-reits** [comment]
g2g when https://github.com/ionic-team/stencil/pull/3863 is all set

File: src/docs/reference/support-policy.md (hunk 1 above)
Comment (rwaskiewicz) [reply src/docs/reference/support-policy.md:78]: We did, I just haven't rebased this since the 4.8 stuff landed ~1 hr ago

Comment (rwaskiewicz) [reply src/docs/reference/support-policy.md:78]: Should be fixed now 👍 https://github.com/ionic-team/stencil-site/pull/939/commits/6ab1e7f6b80848c676f03af43551778bc2103c20

**Review by alicewriteswrongs — APPROVED** [review]
:shipit:

**Pull request merged**